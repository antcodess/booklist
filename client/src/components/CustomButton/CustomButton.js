import React from 'react';

import './CustomButton.scss';

const CustomButton = ({ children, isGoogleSignIn, isRegularSignIn, ...otherProps }) => {
  return (
    <button 
    className={`custom-button ${isGoogleSignIn ? 'google-sign-in' : ''}${isRegularSignIn ? 'regular-sign-in' : ''}`} 
    {...otherProps}>{ children }</button>
  );
};

export default CustomButton;