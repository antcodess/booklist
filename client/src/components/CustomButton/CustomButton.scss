.regular-sign-in {
  margin-bottom: 5px;
  &:hover {
    background-color: black;
    opacity: 0.9;
  }
}

.google-sign-in {
  background-color: #0080DE;

  &:hover {
    opacity: 0.9;
  }
}

@media only screen and (min-width: 950px) {
  .regular-sign-in {
    width: 40%;
  }
}

@media only screen and (min-width: 1100px) {
  .regular-sign-in, .google-sign-in {
    width: 49%;
  }
}