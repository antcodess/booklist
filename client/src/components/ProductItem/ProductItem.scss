.product-item {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  margin-top: 40px;

  .product-item-top {
    display: flex;
    margin-bottom: 60px;

    .product-image {
      width: 50%;

      img {
        max-width: 100%;
        height: auto;
      }
    }

    .product-details {
      width: 50%;
      margin-left: 40px;

      h2 {
        text-align: left;
        margin-bottom: 15px;
        line-height: 1;
      }

      h3 {
        font-family: "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
        color: #AAAAAA;
        font-size: 1.25rem;
        margin-bottom: 15px;
        text-align: left;
      }

      .product-details-text {
        display: flex;
        font-family: Helvetica, Arial, "Lucida Grande", sans-serif;
        color: #AAAAAA;
        margin-bottom: 10px;

        a {
          font-weight: bolder;
          color: #0080DE;

          &:hover {
            color: black;
          }
        }

        .product-details-text-bold {
          font-family: Helvetica, Arial, "Lucida Grande", sans-serif;
          font-weight: bolder;
          color: black;
          padding-right: 5px;
        }
      }

      .product-price {
        margin-top: 15px;
        margin-bottom: 20px;
        display: flex;
        font-family: "Hurme Geometric Sans 3 W03 Bold";
        font-size: 2.5rem;
      }
      
      .custom-button {
        width: 100%;
        margin-bottom: 20px;
        height: 60px;
      }

      .product-share {
        display: flex;

        .share-text {
          font-family: Helvetica, Arial, "Lucida Grande", sans-serif;
          font-weight: bolder;
        }
      }
    }
  }
  .product-item-bottom {
    .product-description {
      h3 {
        text-align: left;
        font-size: 1.20rem;
        margin-bottom: 30px;
      }
      .product-description-text {
        p {
          text-align: left;
          margin-bottom: 20px;
          color: #6A6A6A;
          line-height: 1.6;
        }
      }
    }
  }
}