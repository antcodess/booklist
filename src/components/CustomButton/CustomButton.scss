.regular-sign-in {
  margin-bottom: 5px;
}

.google-sign-in {
  background-color: #0080DE;

  &:hover {
    background-color: black;
  }
}