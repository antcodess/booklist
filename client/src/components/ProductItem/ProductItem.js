import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import ReactHtmlParser from 'react-html-parser';

import BreadCrumbNav from '../BreadCrumbNav/BreadCrumbNav';
import CustomButton from '../CustomButton/CustomButton';

import { addItem } from '../../redux/cart/cart.actions';
import { selectProduct } from '../../redux/shop/shop.selectors';

import './ProductItem.scss';

export const ProductItem = ({ product, addItem }) => {
  const item = product[0];
  const { 
    title,
    imageUrl,
    subTitle,
    author,
    narrator,
    length,
    releaseDate,
    language,
    publisher,
    price,
    description
   } = item;
  const authorURL = encodeURIComponent(author.replace(/[\s]/g, '-').toLowerCase());
  const narratorURL = encodeURIComponent(narrator.replace(/[\s]/g, '-').toLowerCase());
  const publisherURL = encodeURIComponent(publisher.replace(/[\s]/g, '-').toLowerCase());
  return (
    <React.Fragment>
      <BreadCrumbNav title={title} />
      <section className='product-item'>
        <div className='product-item-top'>
          <div className='product-image'>
            <img src={imageUrl} alt='book' />
          </div>
          <div className='product-details'>
            <h2>{title}</h2>
            <h3>{subTitle}</h3>
            <span className='product-details-text'>
              <span className='product-details-text-bold'>By:</span>
              <Link 
                to={`/authors/${authorURL}`}
              >{author}</Link>
            </span>
            <span className='product-details-text'>
              <span className='product-details-text-bold'>Narrated By:</span>
              <Link 
                to={`/authors/${narratorURL}`}
              >{narrator}</Link>
            </span>
            <span className='product-details-text'>
              <span className='product-details-text-bold'>Length:</span>
              {length}
            </span>
            <span className='product-details-text'>
              <span className='product-details-text-bold'>Release Date:</span>
              {releaseDate}
            </span>
            <span className='product-details-text'>
              <span className='product-details-text-bold'>Language:</span>
              {language}
            </span>
            <span className='product-details-text'>
              <span className='product-details-text-bold'>Publisher:</span>
              <Link 
                to={`/authors/${publisherURL}`}
              >{publisher}</Link>
            </span>

            <span className='product-price'>${price}</span>
            <CustomButton onClick={() => addItem(item)}>Add to Cart</CustomButton>
            <div className='product-share'>
              <span className='share-text'>Share:</span>
                <div className='social-icons'>
                  <img src='https://i.ibb.co/LgFdhmx/facebook.png' alt='facebook' />
                  <img src='https://i.ibb.co/fkfzVPc/twitter.png' alt='twitter' />
                  <img src='https://i.ibb.co/RchS8jc/pinterest.png' alt='pinterest' />
                  <img src='https://i.ibb.co/1MdrS6M/google-plus.png' alt='google plus' />
                  <img src='https://i.ibb.co/5K7KhCF/envelope.png' alt='email' />
                </div>
            </div>
          </div>
        </div>
        <div className='product-item-bottom'>
          <div className='product-description'>
            <h3>Publisher Summary</h3>
            <div className='product-description-text'>
              {ReactHtmlParser(description)}
            </div>
          </div>
        </div>
      </section>
    </React.Fragment>
  );
};

const mapStateToProps = (state, ownProps) => ({
  product: selectProduct(ownProps.match.url)(state)
});

export const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item))
});

export default connect(mapStateToProps, mapDispatchToProps)(ProductItem);