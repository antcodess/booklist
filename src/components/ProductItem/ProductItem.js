import React from 'react';

import BreadCrumbNav from '../BreadCrumbNav/BreadCrumbNav';
import CustomButton from '../CustomButton/CustomButton';
import productImage from '../../assets/product.jpg';

import './ProductItem.scss';


const ProductItem = ({ match }) => {
  console.log(match)
  return (
    <React.Fragment>
      <BreadCrumbNav />
      <div className='product-item'>
        <div className='product-item-top'>
          <div className='product-image'>
            <img src={productImage} alt='book' />
          </div>
          <div className='product-details'>
            <h2>Unfu*ck Yourself</h2>
            <h3>Get Out Of Your Head and Into Your Life</h3>
            <span className='product-details-text'><span className='product-details-text-bold'>By:</span><a href='/'>Gary John Bishop</a></span>
            <span className='product-details-text'><span className='product-details-text-bold'>Narrated By:</span><a href='/'>Gary John Bishop</a></span>
            <span className='product-details-text'><span className='product-details-text-bold'>Length:</span>3 hrs and 23mins</span>
            <span className='product-details-text'><span className='product-details-text-bold'>Release Date:</span>08-01-17</span>
            <span className='product-details-text'><span className='product-details-text-bold'>Language:</span>English</span>
            <span className='product-details-text'><span className='product-details-text-bold'>Publisher:</span><a href='/'>Harper Audio</a></span>

            <span className='product-price'>$10</span>
            <CustomButton>Add to Cart</CustomButton>
            <div className='product-share'>
              <span className='share-text'>Share:</span>
            </div>
          </div>
        </div>
        <div className='product-item-bottom'>
          <div className='product-description'>
            <h3>Publisher Summary</h3>
            <div className='product-description-text'>
            <p>New York Times Best Seller</p>

<p>Joining the ranks of The Life-Changing Magic of Not Giving a F*ck, The Subtle Art of Not Giving a F*ck, 
You Are a Badass, and F*ck Feelings is this refreshing, BS-free self-empowerment guide that offers an 
honest, no-nonsense, tough-love approach to help you move past self-imposed limitations.</p>

<p>Are you tired of feeling f*cked up? If you are, Gary John Bishop has the answer. In this straightforward 
handbook, he gives you the tools and advice you need to demolish the slag weighing you down and 
become the truly unf*cked version of yourself. “Wake up to the miracle you are,” he directs. “Here’s 
what you’ve forgotten: You’re a f*cking miracle of being.” It isn’t other people that are standing in your 
way; it isn’t even your circumstances that are blocking your ability to thrive. It’s yourself and the negative 
self-talk you keep telling yourself.</p> 

<p>In Unf*ck Yourself, Bishop leads you through a series of seven assertions:</p> 

<p>I am willing</p>
<p>I am wired to win</p> 
<p>I got this</p> 
<p>I embrace the uncertainty</p> 
<p>I am not my thoughts; I am what I do</p> 
<p>I am relentless</p> 
<p>I expect nothing and accept everything</p> 
<p>Lead the life you were meant to have - Unf*ck Yourself.</p> 

<p>&copy; Gary John Bishop (P)2017 HarperCollins Publishers</p>

            </div>
          </div>
        </div>
      </div>
    </React.Fragment>
  );
};

export default ProductItem;