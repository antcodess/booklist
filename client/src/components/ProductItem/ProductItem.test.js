import React from 'react';
import { shallow } from 'enzyme';
import { ProductItem } from './ProductItem';
import { mapDispatchToProps } from './ProductItem';
import CustomButton from '../CustomButton/CustomButton';
import ReactHtmlParser from 'react-html-parser';

describe('Header Component', () => {
  let wrapper;
  let mockProduct;
  let mockAddItem;
  let mockItem;
  let mockPayload;
  let dummyFunction = jest.fn();

  mockPayload = {
    type: 'ADD_ITEM',
    payload: {
      title: 'Sample Item',
      imageUrl: 'example.com',
      subTitle: 'Sample Subtitle',
      author: 'Sample Author',
      narrator: 'Sample Narrator',
      length: 'Sample Length',
      releaseDate: 'Sample Release Date',
      language: 'English',
      publisher: 'Sample Publisher',
      price: 20,
      description: 'This is a sample description'
    }
  }
  
  beforeEach(() => {
    
    mockItem = {
      title: 'Sample Item',
      imageUrl: 'example.com',
      subTitle: 'Sample Subtitle',
      author: 'Sample Author',
      narrator: 'Sample Narrator',
      length: 'Sample Length',
      releaseDate: 'Sample Release Date',
      language: 'English',
      publisher: 'Sample Publisher',
      price: 20,
      description: 'This is a sample description'
    };

    mockProduct = [mockItem];
    mockAddItem = jest.fn();

    const mockProps = {
      product: mockProduct,
      addItem: mockAddItem
    }
    
    wrapper = shallow(<ProductItem { ...mockProps } />);
  });

  it('should render ProductItem Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addItem when CustomButton is clicked', () => {
    wrapper.find(CustomButton).simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  });

  it('should dispatch addItem action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).addItem(mockItem);
    expect(dispatch.mock.calls[0][0]).toEqual(mockPayload);
  });
});
