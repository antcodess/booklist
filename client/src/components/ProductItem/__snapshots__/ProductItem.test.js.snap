// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render ProductItem Component 1`] = `ShallowWrapper {}`;
